applications/StructuralMechanicsApplication/custom_response_functions/adjoint_conditions/adjoint_semi_analytic_point_load_condition.cpp

```diff
@@ -0,0 +1,212 @@
+
+    void AdjointSemiAnalyticPointLoadCondition::GetValuesVector(Vector& rValues, int Step)
+    {
+        const unsigned int number_of_nodes = GetGeometry().size();
+        const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+        const unsigned int mat_size = number_of_nodes * dim;
+
+        if (rValues.size() != mat_size)
+        {
+            rValues.resize(mat_size, false);
+        }
+
+        for (unsigned int i = 0; i < number_of_nodes; i++)
+        {
+            const array_1d<double, 3 > & Displacement = GetGeometry()[i].FastGetSolutionStepValue(ADJOINT_DISPLACEMENT, Step);
+            unsigned int index = i * dim;
+            for(unsigned int k = 0; k < dim; ++k)
+            {
+                rValues[index + k] = Displacement[k];
+            }
+        }
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+
+    void AdjointSemiAnalyticPointLoadCondition::CalculateSensitivityMatrix(const Variable<double>& rDesignVariable,
+                                            Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY
+
+        KRATOS_ERROR << "There is no scalar design variable available!" << std::endl;
+
+        KRATOS_CATCH( "" )
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+
+    void AdjointSemiAnalyticPointLoadCondition::CalculateSensitivityMatrix(const Variable<array_1d<double,3> >& rDesignVariable,
+                                            Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo)
+    {
+        KRATOS_TRY
+
+        const unsigned int number_of_nodes = GetGeometry().size();
+        const unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+        const unsigned int mat_size = number_of_nodes * dimension;
+
+        if ((rOutput.size1() != mat_size) || (rOutput.size2() != mat_size))
+	        rOutput.resize(mat_size, mat_size, false);
+
+        noalias(rOutput) = ZeroMatrix(mat_size,mat_size);
+
+        if( rDesignVariable == POINT_LOAD )
+        {
+            for(unsigned int i = 0; i < mat_size; ++i)
+                rOutput(i,i) = 1.0;
+        }
+
+        KRATOS_CATCH( "" )
+    }
+
+    //***********************************************************************
+    //***********************************************************************
+
+    int AdjointSemiAnalyticPointLoadCondition::Check( const ProcessInfo& rCurrentProcessInfo )
+    {
+        // verify that the variables are correctly initialized
+        KRATOS_CHECK_VARIABLE_KEY(ADJOINT_DISPLACEMENT);
+        KRATOS_CHECK_VARIABLE_KEY(DISPLACEMENT);
```

Comment:
It is not needed in this class, but in the primal class. 
For the check functions we copied the functions from the primal elements and added the adjoint counterparts of the variables to the `KRATOS_CHECK_VARIABLE_KEY` and replaced them in the `KRATOS_CHECK_DOF_IN_NODE`. This may be overkill but it seems to be the safe way to go.